server/src/main/java/io/druid/server/coordinator/DruidCluster.java

```diff
@@ -86,9 +87,9 @@ private void addRealtime(ServerHolder serverHolder)
   private void addHistorical(ServerHolder serverHolder)
   {
     final ImmutableDruidServer server = serverHolder.getServer();
-    final MinMaxPriorityQueue<ServerHolder> tierServers = historicals.computeIfAbsent(
+    final NavigableSet<ServerHolder> tierServers = historicals.computeIfAbsent(
         server.getTier(),
-        k -> MinMaxPriorityQueue.orderedBy(Ordering.natural().reverse()).create()
+        k -> new TreeSet<>(Collections.reverseOrder())
```

Comment:
This doesn't preserve the collection ordering right? did you mean this to pass a Comparator?